shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,84 @@
+import json
+import math
+import nibabel as nib
+import os
+
+
+def read_nii(niiPath):
+    """
+
+    Args:
+        niiPath (str): Path to the nifti file
+
+    Returns:
+        image:
+        info:
+        jsonData:
+    """
+
+    info = nib.load(niiPath)
+    image = info.get_fdata()
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    jsonPath = niiPath.split('.nii')[0] + '.json'
+
+    if os.path.isfile(jsonPath):
+        jsonData = json.load(open(jsonPath))
+    else:
+        jsonData = []
+        print('Warning: No json file has been found')
+
+    # NOTE: nib.load automatically scales the nifti and replaces scl_inter and scl_slope with 'nan'
+    # See the "Data scaling" section in https://nipy.org/nibabel/nifti_images.html
+    if ('Manufacturer' in jsonData) and (jsonData['Manufacturer'] == 'Siemens') \
```

Comment:
I understand that for now, we have only Siemens, but if there are other manufacturer, it will be troublesome to have a long chain of if for every manufacturer. You can maybe store the function into a dictionary with it's manufacturer as a key. With this, your manufacturer becomes directly callable and is easily added through adding it to your dictionary. Example:
convert_phase = {
    'Siemens' = convert_siemens_phase,
    'Other manufacturer' = convert_other_phase,
}

if ('Manufacturer` in jsonData) and (check_json_image_type(jsonData) == 'phase'):
    image, info = convert_phase[jsonData['Manufacturer']](image, info)

This way, you won't need a lot of ifs, and you can even build a database for these function that would be directly imported.

Reply:
This is good functional-programming advice. Nice.